drivers/scheduler/k8s/k8s.go

```diff
@@ -108,27 +111,34 @@ func (k *k8s) parseSpecs(specDir string, parser func(in interface{}) (interface{
 	}
 
 	var coreSpecs []interface{}
-	for _, file := range fileList {
-		specContents, err := ioutil.ReadFile(file)
+	for _, fileName := range fileList {
+		file, err := os.Open(fileName)
 		if err != nil {
 			return nil, err
 		}
+		defer file.Close()
 
-		obj, _, err := scheme.Codecs.UniversalDeserializer().Decode([]byte(specContents), nil, nil)
-		if err != nil {
-			return nil, err
-		}
+		reader := bufio.NewReader(file)
+		specReader := yaml.NewYAMLReader(reader)
 
-		specObj, err := parser(obj)
-		if err != nil {
-			return nil, err
-		}
+		for {
+			specContents, err := specReader.Read()
+			if err == io.EOF {
+				break
+			}
 
-		if specObj == nil {
-			continue
-		}
+			if len(bytes.TrimSpace(specContents)) > 0 {
+				obj, _, err := scheme.Codecs.UniversalDeserializer().Decode([]byte(specContents), nil, nil)
+				if err != nil {
+					return nil, err
+				}
 
-		coreSpecs = append(coreSpecs, specObj)
+				specObj, err := parser(obj)
+				if specObj != nil {
```

Comment:
Catch the parser error here and return if not nil.

Reply:
The parser does not throw any error right now. Probably we can get rid of the error from the parse method.